client/multi-currency-setup/tasks/store-settings-task/index.js

```diff
@@ -65,10 +70,18 @@ const StoreSettingsTask = () => {
 
 	const handleContinueClick = () => {
 		setPending( true );
+
+		if ( ! isMultiCurrencyEnabled ) {
+			updateIsMultiCurrencyEnabled( true );
+			saveSettings();
+		}
+
 		submitStoreSettingsUpdate(
 			isAutomaticSwitchEnabledValue,
-			isStorefrontSwitcherEnabledValue
+			isStorefrontSwitcherEnabledValue,
+			! isMultiCurrencyEnabled
```

Comment:
That's correct. We just want to suppress the `submitStoreSettingsUpdate` toast if we are also triggering the `saveSettings` function to update the multi-currency setting, as it will trigger a toast as well.

The third argument of `submitStoreSettingsUpdate` is `suppressNotices` which aims to update the multi-currency store settings silently.

This is how the toasts look without suppressing one of them:
<img width="250" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/16882226/c633f44a-9a27-4c08-8e1b-214e53696cf2">

Reply:
If multi-currency is enabled, then we don't need to call `saveSettings` or suppress the toast from `submitStoreSettingsUpdate`, that's why they're related.

Reply:
Here is the suppressing mechanism on `submitStoreSettingsUpdate`:

https://github.com/Automattic/woocommerce-payments/blob/0ed79c06c0a524b42234838f5dbb85d7c3ced1f1/client/data/multi-currency/actions.js#L119-L150

Reply:
Do you think the code would be clearer if we moved the suppress notice mechanism to the `saveSettings` function?

That's how it would look:

```js
	const handleContinueClick = () => {
		setPending( true );

		if ( ! isMultiCurrencyEnabled ) {
			updateIsMultiCurrencyEnabled( true );
			saveSettings( true );
		}

		submitStoreSettingsUpdate(
			isAutomaticSwitchEnabledValue,
			isStorefrontSwitcherEnabledValue
		);

		setPending( false );
		setCompleted( true, 'setup-complete' );
	};
```

Reply:
Thanks for the explanation. It makes sense. I guess I was just not fully focused when I was looking at it the first time and became a bit confused as a result. I think it is fine to keep the current implementation.